### PR TITLE
Fix Bug, Speed up

### PR DESCRIPTION
**Pull request by afrasier**
Speeds up File F ETL by only pulling subawards we have the ability to
match and load.

Fixes a bug with FAIN lookup.

Updates tests

### CODE REVIEW

File: usaspending_api/etl/subaward_etl.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
                                          AND fsrs_procurement.idv_reference_number = award_procurement.parent_award_id
        LEFT OUTER JOIN fsrs_subcontract  ON fsrs_subcontract.parent_id = fsrs_procurement.id
 WHERE  award_procurement.submission_id = %s
+       AND award_procurement.piid IN %s
```
Comment (bsweger) [inline usaspending_api/etl/subaward_etl.py]: I haven't been following along--do we need to care about parent award id for the purposes of this lookup?

File: usaspending_api/etl/subaward_etl.py [hunk 2]
```diff
@@ -26,6 +27,7 @@
        LEFT OUTER JOIN award_financial_assistance ON fsrs_grant.fain = award_financial_assistance.fain
        LEFT OUTER JOIN fsrs_subgrant ON fsrs_grant.id = fsrs_subgrant.parent_id
 WHERE  award_financial_assistance.submission_id = %s
+       AND award_financial_assistance.fain IN %s
```
Comment (bsweger) [inline usaspending_api/etl/subaward_etl.py]: Similar question to above--do we care about URI here?

File: usaspending_api/etl/subaward_etl.py (hunk 1 above)
Comment (afrasier) [inline usaspending_api/etl/subaward_etl.py]: No, we check for a matching parent award ID at the bottom; this is just a small check that ensures we only pull Records we have the _possibilty_ of matching.

**Comment by afrasier** [comment]
I do have tests for submission/piid/fain/etc. matching, making sure they attach properly. They were part of the previous PR.

File: usaspending_api/etl/subaward_etl.py [hunk 3]
```diff
@@ -68,6 +65,7 @@ def load_subawards(submission_attributes, db_cursor):
         # a matching parent award id, piid, and submission attributes
         award = Award.objects.filter(transaction__submission=submission_attributes,
                                      transaction__contract_data__piid=row['piid'],
+                                     transaction__contract_data__isnull=False,
```
Comment (bsweger) [inline usaspending_api/etl/subaward_etl.py]: If the filters are AND'd together, would `transaction__contract_data__isnull=False` always be true if `transaction__contract_data__piid=row['piid']`?

Comment (afrasier) [inline usaspending_api/etl/subaward_etl.py]: Nope! OneToOne relationships do this weird thing where even though contract_data is null, it still lets you check contract_data__piid, and returns null. Just making sure to be explicit here.

Comment (afrasier) [inline usaspending_api/etl/subaward_etl.py]: i.e. `Transaction.objects.filter(contract_data__isnull=True, contract_data__piid__isnull=True)` is perfectly valid (which is kind of weird, but... django)

File: usaspending_api/etl/subaward_etl.py [hunk 4]
```diff
@@ -68,6 +65,7 @@ def load_subawards(submission_attributes, db_cursor):
         # a matching parent award id, piid, and submission attributes
         award = Award.objects.filter(transaction__submission=submission_attributes,
```
Comment (bsweger) [inline usaspending_api/etl/subaward_etl.py]: I know this code is already in dev, but something I found helpful when doing various submission loader fixes is make a discrete method on the model (in this case, `Award`) for these types of matches. That made it a lot easier to unit test the edge cases with good old model mommy.

File: usaspending_api/etl/subaward_etl.py (hunk 3 above)
Comment (bsweger) [inline usaspending_api/etl/subaward_etl.py]: We discussed offline. All is well.

File: usaspending_api/etl/tests/test_subaward_etl.py [hunk 5]
```diff
@@ -27,7 +27,9 @@ def test_subaward_etl_fixture():
 
     # Transaction with subaward, by FAIN
     txn2 = mommy.make(Transaction, award=prime_award_2, submission=submission)
-    mommy.make(TransactionAssistance, transaction=txn2, fain=test_fain, uri=test_uri)
+    txn3 = mommy.make(Transaction, award=prime_award_2, submission=submission)
+    mommy.make(TransactionAssistance, transaction=txn2, fain=test_fain, uri=None)
+    mommy.make(TransactionAssistance, transaction=txn3, fain=None, uri=test_uri)
```
Comment (bsweger) [inline usaspending_api/etl/tests/test_subaward_etl.py]: Thanks for adding this!
